src/main/resources/org/jenkinsci/plugins/workflow/support/steps/stash/StashStep/help.html

```diff
@@ -1,6 +1,7 @@
 <div>
     Saves a set of files for use later in the same build, generally on another node/workspace.
     Stashed files are not otherwise available and are generally discarded at the end of the build.
+    Stashes are not available to be used cross jobs. For this you need to use <code>archiveArtifacts</code>/<code>copyArtifacts</code>.
```

Comment:
`copyArtifacts` comes from a non-core plugin, so I don't think we want to recommend it directly. Maybe tweak the wording to something like this:
```suggestion
    If you want artifacts to persist after the build, consider using <a href="https://jenkins.io/doc/pipeline/steps/core/#-archiveartifacts-%20archive%20the%20artifacts"><code>archiveArtifacts</code></a> instead.
```

Reply:
I like this variant more. What do you think?
```suggestion
    Stashes are not available to be used cross jobs. If you want artifacts to persist after the build, consider using <a href="https://jenkins.io/doc/pipeline/steps/core/#-archiveartifacts-%20archive%20the%20artifacts"><code>archiveArtifacts</code></a> instead.
```

Reply:
Compromise:
```suggestion
    Stashes from a particular pipeline run are not available in other runs or other pipelines.
    If you want artifacts to persist after the build, consider using 
    <a href="https://jenkins.io/doc/pipeline/steps/core/#archiveartifacts-archive-the-artifacts"
    <code>archiveArtifacts</code></a> instead.
```